File: mengshi/topic_pub_py/topic_pub_py/pub_vel.py
#ros2 pkg create topic_pub_py --build-type ament_python --dependencies rclpy geometry_msgs --node-name pub_vel 
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist

class PubVel(Node):
    def __init__(self):
        super().__init__('pub_vel')
        self.declare_parameter('linear_x', 0.0)
        self.declare_parameter('angular_z', 0.0)
        self.twist_pub=self.create_publisher(Twist, '/cmd_vel', 10)
        #qos 10
        self.timer=self.create_timer(1.0,self.on_timer)
        
    def on_timer(self):
        #orgnization of Twist
        twist=Twist()
        twist.linear.x=self.get_parameter('linear_x').value
        twist.angular.z=self.get_parameter('angular_z').value
        self.twist_pub.publish(twist)
            
def main(args=None):
    rclpy.init(args=args)
    pub_vel=PubVel()
    rclpy.spin(pub_vel)
    pub_vel.destroy_node()
    rclpy.shutdown()
        

File: mengshi/topic_pub/src/pub_vel.cpp
// ros2 pkg create topic_pub --build-type ament_cmake --dependencies rclcpp geometry_msgs --node-name pub_vel
//circulation publish vel msg
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"
#include <chrono>
using namespace std::chrono_literals;
class PubVel : public rclcpp::Node
{
public:
    PubVel():Node("pub_vel")
    {
      //dynimic param init, templeate type is double
        this->declare_parameter<double>("linear_x",0.0);
        this->declare_parameter<double>("angular_z",0.0);
        publisher_ = this->create_publisher<geometry_msgs::msg::Twist>("/cmd_vel",10);
        timer_ = this->create_wall_timer(100ms,std::bind(&PubVel::timer_callback,this));
    }

private:
    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
    void timer_callback()
    {
        geometry_msgs::msg::Twist twist;
        twist.linear.x = this->get_parameter("linear_x").as_double();
        twist.angular.z = this->get_parameter("angular_z").as_double();
        publisher_->publish(twist);
        RCLCPP_INFO(this->get_logger(),"Publishing: linear.x = %f, angular.z = %f",msg_->linear.x,msg_->angular.z);
    }
}
int main(int argc,char **argv)
{
    //init ros2
    rclcpp::init(argc,argv);
    //spin and input obj ptr
    rclcpp::spin(std::make_shared<PubVel>());
    /*
    //if you use base obj programming
    auto node = rclcpp::Node::make_shared("pub_vel");
    auto publisher = node->create_publisher<geometry_msgs::msg::Twist>("/cmd_vel",10);
    auto msg = std::make_shared<geometry_msgs::msg::Twist>();
    msg->linear.x = 1.0;
    msg->angular.z = 1.0;
    rclcpp::WallRate loop_rate(1);
    while(rclcpp::ok())
    {
        publisher->publish(msg);
        RCLCPP_INFO(node->get_logger(),"Publishing: linear.x = %f, angular.z = %f",msg->linear.x,msg->angular.z);
        rclcpp::spin_some(node);
        loop_rate.sleep();
    }
    */
    //release resources
    rclcpp::shutdown();
    return 0;
}